lib/factory.ts

```diff
@@ -11,7 +11,9 @@ import { FactoryBuilder } from './builder';
 const SEQUENCE_START_VALUE = 1;
 
 export class Factory<T, I = any> {
-  private nextId: number = SEQUENCE_START_VALUE;
+  // id is an object so it is shared between extended factories
+  private id: { value: number } = { value: SEQUENCE_START_VALUE };
```

Comment:
minor: I'd probably go with something related to `sequence` instead of `id` for consistency.

Reply:
Thanks for this suggestion. `sequence.value` was my first try, but I couldn't use that naming since sequence is already a method on `Factory`.